src/ast/mod.rs

```diff
@@ -3954,6 +3954,33 @@ impl fmt::Display for SearchModifier {
     }
 }
 
+/// A cols definition (i.e. `cols(view_schema name, view_name name, col_name name, col_type varchar, col_num int)`
+/// when used with redshift pg_get_late_binding_view_cols/pg_get_cols)
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct ColsDefinition {
```

Comment:
Is there any reason not to use the existing `ColumnDef`? 

https://github.com/sqlparser-rs/sqlparser-rs/blob/61c661c234518ae02ae2c50d317e4640e4b90e26/src/ast/ddl.rs#L479-L486

I think that would make this PR significantly shorter as well as supporting other options

What do you think?

Reply:
I think that the name is misleading. I should have named it e.g. `TableAliasDefinition`, because it defines what will be a result table, what column names it will have.
Like in the example:
```
select * from pg_get_late_binding_view_cols() cols(view_schema name, view_name name, col_name name, col_type varchar, col_num int);
view_schema | view_name | col_name   | col_type                    | col_num
------------+-----------+------------+-----------------------------+--------
public      | sales_lbv | salesid    | integer                     |       1
public      | sales_lbv | listid     | integer                     |       2
...
```
But I think it's a good idea to use `ColumnDef` inside the vector, instead of vector of `IdentPair`s.

What's your opinion?

Reply:
> I think that the name is misleading. I should have named it e.g. TableAliasDefinition, because it defines what will be a result table, what column names it will have.

Sounds like a good idea to me

> But I think it's a good idea to use ColumnDef inside the vector, instead of vector of IdentPairs.


👍

Reply:
I've tried using `ColumnDef` and failed because `name` is not a column type. It's strange, redshift doesn't define it as a keyword nor a type, but it's used in such a case. I think it's not a good idea to add it as additional type, so I'll rather back to `IdentPair`, or is there a better way to handle it?

Reply:
So I looked at the docs for

https://docs.aws.amazon.com/redshift/latest/dg/PG_GET_GRANTEE_BY_IAMROLE.html

I see in the example it has something like

```sql
select grantee, grantee_type, cmd_type 
FROM 
  pg_get_grantee_by_iam_role('arn:aws:iam::123456789012:role/Redshift-S3-Write') 
  res_grantee(grantee text, grantee_type text, cmd_type text) 
ORDER BY
 1,2,3;
```

I think this PR is related to this bit of the query:

```
  res_grantee(grantee text, grantee_type text, cmd_type text) 
```

I have several confusions:
1. Are you sure this construct is specific to the system information functions, or is a more general mechanism (it look like a more general mechanism that expands out a tuple to a table to me)
2. If it is a view definition, I would expect it to be parsed as ident, variable type not as just generic idents


Maybe someone could do some more research into what exactly this syntax construct in Redshift is called and what its rules are

Reply:
1. You're right, it looks like a more general mechanism. I've checked the documentation, but haven't found any general syntax description for it.
2. So if we have an additional `name` type, then it has to be added to the `DataType`, right?